File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/action/KeepAliveAction.java
package org.emulinker.kaillera.controller.v086.action;

import javax.inject.Inject;
import javax.inject.Singleton;
import org.emulinker.kaillera.controller.v086.V086Controller.V086ClientHandler;
import org.emulinker.kaillera.controller.v086.protocol.*;

@Singleton
public class KeepAliveAction implements V086Action<KeepAlive> {
  private static final String DESC = "KeepAliveAction";

  private int actionCount = 0;

  @Inject
  KeepAliveAction() {}

  @Override
  public int getActionPerformedCount() {
    return actionCount;
  }

  @Override
  public String toString() {
    return DESC;
  }

  @Override
  public void performAction(KeepAlive message, V086ClientHandler clientHandler)
      throws FatalActionException {
    actionCount++;

    if (clientHandler.getUser() == null)
      throw new FatalActionException("User does not exist: KeepAliveAction!");

    clientHandler.getUser().updateLastKeepAlive();
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/action/KeepAliveAction.kt
package org.emulinker.kaillera.controller.v086.action

import javax.inject.Inject
import javax.inject.Singleton
import org.emulinker.kaillera.controller.v086.V086ClientHandler
import org.emulinker.kaillera.controller.v086.protocol.KeepAlive

@Singleton
class KeepAliveAction @Inject internal constructor() : V086Action<KeepAlive> {
  override var actionPerformedCount = 0
    private set

  override fun toString() = "KeepAliveAction"

  @Throws(FatalActionException::class)
  override suspend fun performAction(message: KeepAlive, clientHandler: V086ClientHandler) {
    actionPerformedCount++
    if (clientHandler.user == null)
        throw FatalActionException("User does not exist: KeepAliveAction!")
    clientHandler.user.updateLastKeepAlive()
  }
}
